lib/pool.js

```diff
@@ -82,14 +92,12 @@ class Pool extends EventEmitter {
       }
     }
 
-    this[kOnConnect] = function onConnect (client) {
-      pool[kConnected]++
-      pool.emit('connect', client)
+    this[kOnConnect] = function onConnect (origin, targets) {
```

Comment:
Wouldn't it be better to pass `onConnect` as an option? That way there would be no need for an `EventEmitter`.